### PR TITLE
chore: migrate phpunit v9 => v10

### PR DESCRIPTION
**Pull request by Chris53897**
I am unsure about i did this test-replacement correct. Please have a look.
testRender() on https://github.com/KnpLabs/KnpMenuBundle/compare/master...Chris53897:KnpMenuBundle:feature/phpunit-10?expand=1#diff-b293cb7a5c5500f23a6c3f9fa8ff41dc1f2691fce1c7606c78f9b081041defa0R105

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 .*.cache
 composer.lock
+.phpunit.result.cache
+.phpunit.cache
```
Comment (garak) [inline .gitignore:4]: these two cases are already covered above. Please remove

File: composer.json [hunk 2]
```diff
@@ -24,7 +24,7 @@
         "symfony/framework-bundle": "^3.4 | ^4.4 | ^5.0 | ^6.0"
     },
     "require-dev": {
-        "phpunit/phpunit": "^9.5",
+        "phpunit/phpunit": "^9.6 | ^10.1",
```
Comment (garak) [inline composer.json:27]: it doesn't make much sense to keep two versions for a dev tool like PHPUnit. Let's drop version 9

File: phpunit.xml.dist [hunk 3]
```diff
@@ -2,18 +2,22 @@
 <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          colors="true"
          bootstrap="vendor/autoload.php"
-         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.5/phpunit.xsd">
-  <coverage>
+         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/10.1/phpunit.xsd">
+  <coverage/>
```
Comment (garak) [inline phpunit.xml.dist:6]: I guess we can remove this empty tag, right?

File: phpunit.xml.dist [hunk 4]
```diff
@@ -2,18 +2,22 @@
...
+
+  <testsuites>
```
Comment (garak) [inline phpunit.xml.dist:8]: Can this tag be kept in the previous position? Just to minimize changes

**Review by garak — CHANGES_REQUESTED** [review]

File: composer.json (hunk 2 above)
Comment (derrabus) [reply composer.json:27]: PHPUnit 10 requires PHP 8.1. As long as we need to run tests on PHP 7, we need PHPUnit 9. And dropping support for PHP 7 is probably out of scope for this PR?

File: phpunit.xml.dist [hunk 5]
```diff
@@ -2,18 +2,22 @@
 <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          colors="true"
          bootstrap="vendor/autoload.php"
-         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.5/phpunit.xsd">
-  <coverage>
+         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/10.1/phpunit.xsd">
```
Comment (derrabus) [inline phpunit.xml.dist:5]: PHPUnit 9 won't understand this configuration file. You need to stick with the (deprecated) format as long as you run tests on PHPUnit 9.

File: composer.json (hunk 2 above)
Comment (garak) [reply composer.json:27]: good point

File: phpunit.xml.dist (hunk 5 above)
Comment (Chris53897) [reply phpunit.xml.dist]: I have reverted the config-file

File: phpunit.xml.dist (hunk 4 above)
Comment (Chris53897) [reply phpunit.xml.dist]: I have reverted the config-file. (see below)

File: .gitignore (hunk 1 above)
Comment (Chris53897) [reply .gitignore]: thanks, i did not see this. reverted

**Review by derrabus — APPROVED** [review]

**Review by garak — APPROVED** [review]

**Pull request merged**